Sources/Fuzzilli/Core/CodeTemplate.swift

```diff
@@ -0,0 +1,100 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+public class CodeTemplate {
+    public let name: String
+
+    public var stats = ProgramGeneratorStats()
+
+    private let f: (ProgramBuilder) -> ()
+
+    init(_ name: String, _ f: @escaping (_: ProgramBuilder) -> ()) {
+        self.name = name
+        self.f = f
+    }
+
+    func run(in b: ProgramBuilder) {
+        f(b)
+    }
+
+    /// Generate an array of random function signatures
+    static func generateRandomFunctionSignatures(forFuzzer fuzzer: Fuzzer, n: Int) -> [FunctionSignature] {
+        var signatures = [FunctionSignature]()
+        signatures.append(CodeTemplate.generateSignature(forFuzzer: fuzzer, n: Int.random(in: 0...3)))
```

Comment:
Should probably be `n: n`? :)

Reply:
Oh wow actually I wanted to return multiple signatures, the whole loop is missing :D